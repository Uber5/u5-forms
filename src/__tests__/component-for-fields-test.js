import React from 'react'

import { componentForFields } from '../'

describe('componentForFields', () => {
  it('constructs fields for each field type', () => {
    const fields = [
      { name: 'someText', label: 'Label for multiline', type: 'multiline' },
      { name: 'someDate', label: 'Label for a date', type: 'date' },
      { name: 'someField', label: 'Some field without type' },
      { name: 'someTextField', label: 'Label for text field', type: 'text' },
      { name: 'someNameField', label: 'Label for name field', type: 'name' },
      { name: 'someEmailField', label: 'Label for email field', type: 'email' },
      { name: 'someIdnoField', label: 'Label for idno field', type: 'idno' },
      { name: 'someBooleanField', label: 'Label for boolean field', type: 'boolean' },
    ]
    const Component = componentForFields(fields)
    const rendered = render(<Component />)
    expect(rendered).toMatchSnapshot()
  })
})
