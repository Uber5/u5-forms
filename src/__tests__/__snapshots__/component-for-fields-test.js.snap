exports[`componentForFields constructs fields for each field type 1`] = `
<p>
  components
</p>
`;
